### ISSUE TITLE
Computype Three Label Set - 1005544-1

### ISSUE THREAD

**Issue by YvonneKester** [ExLibrisGroup/spineomatic-cloudapp#69]
Hello,

I'm having trouble visualizing how to create this template/layout for SpineOMatic and am hoping you can give me some help. This is a three-label set, with a spine label on one side and two "pocket" labels on the other side. I was looking at your directions for the other Computype labels, and maybe I'm too tired but I just can't visualize and extrapolate to the other style. The dimensions I'm working with are:
Three Label Set - 1005544-1
Label 1 - 0.9" x 1.56"
Label 2 - 2.9" x 0.8125"
Label 3 - 2.9" x 0.8125"

And I have a drawing, attached here. Can you give me some guidance?
Many thanks!
[part 1005544-1.pdf](https://github.com/ExLibrisGroup/spineomatic-cloudapp/files/8711623/part.1005544-1.pdf)

Yvonne

**Comment by mgobat** [comment]
This is untested, as we don't have a dedicated label printer or this label stock. If its close to what you need, it might need some tweaking. And if its not even close, we'll try again.

Create the layout with these settings:
- Measurement units: Inches
- Top margin: .125
- Left margin: .125
- Page width: 4.175
- Orientation: Portrait
- Label width: 3.925
- Label height: 1.625
- Horizontal gap: .125
- Vertical gap: .125
- Labels per page: 1
- Left padding: 0

Create the template:
- Create a table with 3 columns and 2 rows.
- Select both rows in the first column, right click, select "Cell > Merge cells".
- Select the cell in the first row, right click, select "Cell > Cell properties" and set width = .9in height = .78in
- Select both cells in the second column, right click, select "Cell > Cell properties" and set width = .125in and height = 1.625in
- Select both cells in the third column, right click, select "Cell > Cell properties" and set width = 2.9in and height = .8125in
- Insert the fields to print in only the first and third columns in the appropriate row.
- First column is your spine label.
- Third column is your two pocket labels.
- Second column represents the gaps between the labels; leave empty.

Please let us know how it works...or doesn't. Thanks.

**Comment by YvonneKester** [comment]
Hello,

Thank you very much!
I've been tweaking this, and have gotten to the point where it tries to print three labels, but for some reason the bottom of the spine label and second pocket label isn't printing. I'm attaching a PDF print of where I'm at.

The cell properties width is in percentage, so I calculated that, and the height is in pixels, so I calculated that.

For the spine label I have: Width = 22%; Height = 156px Horizontal align = Left and Vertical align = Middle

For the top pocket label I have: Width = 69%; Height = 78px; Horiz align = Left; Vert align = Middle
For the bottom pocket label I have: Width = 69%; Height = 78px; Horiz align = Left; Vert align = Middle

The layout is as you specified above.

Do you have any ideas about why it is cutting off the bottom of the labels?
Thanks again,
Yvonne
[Three-LabelExample.pdf](https://github.com/ExLibrisGroup/spineomatic-cloudapp/files/8733025/Three-LabelExample.pdf)
.

**Comment by YvonneKester** [comment]
Hi again--I figured out that I can add the dimensions in inches, and that has helped! (See attached.) One thing I am wondering is, is it possible to add line breaks in the call number for the spine label, but not on the pocket labels? It seems to allow only one global setting for this.
[Second try in inches.pdf](https://github.com/ExLibrisGroup/spineomatic-cloudapp/files/8734134/Second.try.in.inches.pdf)

**Comment by YvonneKester** [comment]
Update:  I had the campus try the new settings, but their printout looked much worse than mine! I'm attaching. It seems like the table really isn't the same size as the actual labels?
![IMG_1176](https://user-images.githubusercontent.com/105752712/169546629-4ba28b0d-d08e-4d2e-bc0b-60b23ac00932.jpg)

**Comment by YvonneKester** [comment]
@mgobat,
Yes, I was so excited when I saw that, but also yes, I was printing to PDF and using Letter size paper. They of course are using the custom size. It would be great to be able to set up their printer on my machine, but I don't think that's possible, so I keep making adjustments and having them test. That said, when I see their printouts, it doesn't look like the table I've set in SpineOMatic is actually the size it's supposed to be.

Question: I keep making adjustments, and figured out that I could put in inches by adding "in" to the end of the number, like "4.175in". However, many times I'll go back in and my setting has been changed to pixels. Does the program support inches, or should I go back to % and px?

If we could get these printing that would be a great step forward, regardless of call number format, but I will ask them about it. If it is necessary I will put in an enhancement request.

Thanks so much!

**Comment by mgobat** [comment]
@YvonneKester, it should support "in" appended to a numeric size. I think it defaults back to % or px if you add or delete a row or column, though. I'll have to check; it is a third party library.

The intent is that the combined template and layout should appropriately match the label being used, provided all the setting sizes are correct for the target label. Although this is the first time we are trying 3 labels...

**Comment by YvonneKester** [comment]
Hi @mgobat,
I've been away on staycation for the past couple weeks, so just getting back to this. I made some changes and got some better results, but it still isn't "printing" correctly. One problem seems to be an inability to hold my changes to inches. I made a short screencast to show you what I'm seeing. Hope it is helpful! https://www.screencast.com/t/IHGXFs60q2EG

**Comment by YvonneKester** [comment]
@mgobat Thank you for your response--I, too, have been working on other things. Last night I was able to get a fairly good test view, and I put it out for the campus to test, but they can't open the app now. (Hence my recent issue.) I am continuing to work on getting this configured, and will let you know how it goes.

**Comment by YvonneKester** [comment]
@mgobat Hello,
I did more configuring, and I'm just stymied as to how to proceed. I'm attaching screenshots of what I see in the print preview, and what actually comes out. I'm stuck. The IT department for this campus refuses to support the program on their computers, and I can't get the cloud app to work. Only one user is able to successfully print labels using the SpineOMatic program on their computer. Do you have any suggestions for how I can get them moving again? 

Many thanks for all your help.
Yvonne
![ActualOutput](https://user-images.githubusercontent.com/105752712/175552
![PrintPreviewinAlma](https://user-images.githubusercontent.com/105752712/175552561-831ffa55-7258-477a-ac5f-d5acfb56bb4e.JPG)
551-7366d463-80a9-43e2-b878-9f88d27daf71.JPG)

**Comment by YvonneKester** [comment]
![ActualOutput](https://user-images.githubusercontent.com/105752712/175552763-c2bec087-228c-440d-a140-49b442bb6dc8.JPG)

**Comment by YvonneKester** [comment]
Hi,
Were you able to get any information from the third-party table designer?
Thanks,
Yvonne

**Comment by YvonneKester** [comment]
Hi @mgobat,
Any update on this?
Thanks,
Yvonne

**Comment by mgobat** [comment]
Hi @YvonneKester. I've been struggling with getting my test case to revert from inches to pixels. I stepped away from it for awhile because of that. Today, I'm still not able to replicate the behavior you are seeing with inches switching to pixels. I also did not find anything on the table tool developer site reporting such an issue. But it is clearly happening for you. Could you provide complete details of your template (table dimensions, row/column dimensions, fields in each cell) to see if I can replicate it? Thanks.

**Comment by YvonneKester** [comment]
Hi @mgobat , thanks. I was working on it Monday and I think I found the main issue, which also seems to have an effect on measures changing from inches to pixels and percent. I start out with two rows, three columns, then merge the two rows in the first column to make the spine label. the middle column is the space between the spine label and the two pocket labels. I discovered the rows that are merged are still retaining the settings from the rows that are not merged. Every time I change the measurements of the spine label cell, it changes the measurements of the pocket label cells, and vice versa.

My layout:
Measurement in inches, Portrait orientation.
Top margin: 0.125in
Left margin: 0.125in
Page width: 4.175in
Label width: 3.925in
Label height 1.625in
Horizontal gap: 0.125in
Vertical gap: 0.125in
Labels per page: 1
Left padding 0

Template:
Spine label cell properties:
Width: 0.9in
Height: 1.625in
Cell type: cell
Scope: none
Horiz align: Left
Vertical align: middle

Spine label Row Properties:
Row type: Body ( I did play with these a bit, but they didn't seem to do anything)
Alignment: Left
Height: 1.625

Middle column (spacer)
Width: 0.125in
Height: 1.625in
Cell type: Cell
Scope: none
Horiz and vert align: none

Right column--the pocket labels (I have tried leaving these as two rows and also merging them. Right now they are merged)
Width: 2.9in
Height: 1.625in
Cell Type: Cell
Scope:None
Horiz align: left
Vert align: top (so I could try spacing the two sets of type within one big cell)

When the rows for the pocket labels were separate, I made the height of each cell 0.8125in.

I hope this helps. Thanks!

**Comment by mgobat** [comment]
@YvonneKester, thank you for the detailed steps. I have replicated the issue of inches reverting to pixels. Now to figure out what to do about it...

Not that this is a solution, but we exposed the ability to directly edit the HTML of the template. Click the "<>" icon that appears in the toolbar in front of the tables icons.

**Comment by YvonneKester** [comment]
Omg, now I'm trying to build the table. ;-) Two side-by-side tables, even.

**Comment by YvonneKester** [comment]
Hi @mgobat,

I have been playing with the code, and it gives me hope, but I can't seem to get the left (spine) label to stay put as several labels print one after another. Any pointers on how to get a fixed position? I'm attaching my table html as a text file, and here is a screencast of what I mean: https://www.screencast.com/t/BAEIRnCf
[3-LabelTablehtml.txt](https://github.com/ExLibrisGroup/spineomatic-cloudapp/files/9427732/3-LabelTablehtml.txt)
Thanks!
Yvonne
